Userland/Utilities/nc.cpp

```diff
@@ -114,7 +114,7 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         TRY(Core::System::getsockname(listen_fd, (struct sockaddr*)&sin, &len));
 
         if (verbose)
-            warnln("waiting for a connection on {}:{}", inet_ntop(sin.sin_family, &sin.sin_addr, addr_str, sizeof(addr_str) - 1), ntohs(sin.sin_port));
+            warnln("waiting for a connection on {}:{}", inet_ntop(sin.sin_family, &sin.sin_addr, addr_str, sizeof(addr_str)), ntohs(sin.sin_port));
```

Comment:
Does `inet_ntop` guarantee null-termination? Otherwise, that is most likely the reason for the `-1`.

Granted, for that to work we should zero-initialize that buffer first. And we also should check for errors.

Reply:
The not checking for errors was the reason why it went under the radar for so long. `errno` was set to `28` as it should. What would be the best approach for fixing this once and for all (it now works, but maybe breaks again some day)?

Reply:
> The buffer dst must be at least INET_ADDRSTRLEN bytes long.

It now is exactly as long as `INET_ADDRSTRLEN`

Reply:
> What would be the best approach for fixing this once and for all (it now works, but maybe breaks again some day)?

Either we should use an explicit length to pass the string to the logging function so that we don't need the null termination, or we resize the buffer to one byte bigger than needed (i.e. `INET_ADDRSTRLEN + 1`) and make sure that it's zero-initialized.

Reply:
How would one pass a length to `warnln`? Isn't that just putting the variables into `{}` and printing it so `stderr`?

Reply:
So, I just checked the code.

`inet_ntop` has 2 branches:

`AF_INET` -> Uses `snprintf()` to fill the buffer. It will always terminate  with `\0`.
`AF_INET6` -> Uses `DeprecatedString::copy_characters_to_buffer()` that will always terminate with `\0`.